umpost/um2netcdf.py

```diff
@@ -92,6 +93,34 @@ class UnsupportedTimeSeriesError(PostProcessingError):
     pass
 
 
+# TODO: Move this to a separate helper file?
+class EnumAction(argparse.Action):
+    """
+    Argparse action for handling Enums.
+    It automatically produces choices based on the Enum values.
+    """
+
+    def __init__(self, **kwargs):
+        # Pop the 'type' keyword
+        enum_type = kwargs.pop("type", None)
+        # Ensure an Enum subclass is provided
+        if not enum_type or not issubclass(enum_type, Enum):
+            raise TypeError(
+                f"The 'type' keyword must be assigned to Enum (or any Enum subclass) when using {self.__class__.__name__}."
+            )
+        # Generate choices from the Enum values
+        kwargs.setdefault("choices", tuple(e.value for e in enum_type))
+        # Call the argparse.Action constructor with the remaining keyword arguments
+        super().__init__(**kwargs)
+        # Store Enum subclass for use in the __call__ method
+        self._enum = enum_type
+
+    def __call__(self, parser, namespace, value, option_string=None):
+        # Convert value to the associated Enum member
+        member = self._enum(value)
+        setattr(namespace, self.dest, member)
+
+
```

Comment:
That is a solution as well, but the Action is more general and can give us more control if we need it. 
Also, it can be used for any other enumeration in the same exact way.

Reply:
Sounds good!

Reply:
Changes can be found in [43fdb17ac9dbcb226e28bbc8093dc99db51e623b](https://github.com/ACCESS-NRI/um2nc-standalone/pull/160/commits/43fdb17ac9dbcb226e28bbc8093dc99db51e623b)